huggingface_sb3/push_to_hub.py

```diff
@@ -7,30 +7,309 @@
 
 from huggingface_hub import HfApi, HfFolder, Repository
 
+import stable_baselines3
+from stable_baselines3 import *
+from stable_baselines3.common.vec_env import *
+from stable_baselines3.common.env_util import *
+from stable_baselines3.common.evaluation import evaluate_policy
+
+import pickle5
+import json
+import gym
+import zipfile
+
+from subprocess import call
+
 README_TEMPLATE = """---
 tags:
 - deep-reinforcement-learning
 - reinforcement-learning
-- stable-baselines3
 ---
 # TODO: Fill this model card
 """
 
-def _create_model_card(repo_dir: Path):
+
+def _generate_config(model, repo_local_path):
+    """
+    Generate a config.json file containing information about the agent and the environment
+    :param model: name of the model zip file
+    :param repo_local_path: path of the local repository
+    """
+    unzipped_model_folder = model
+
+    # Check if the user forgot to mention the extension of the file
+    if model.endswith('.zip') is False:
+        model += ".zip"
+
+    # Step 1: Unzip the model
+    with zipfile.ZipFile(Path(repo_local_path) / model, 'r') as zip_ref:
+        zip_ref.extractall(Path(repo_local_path) / unzipped_model_folder)
+
+    # Step 2: Get data (JSON containing infos) and read it
+    with open(Path.joinpath(repo_local_path, unzipped_model_folder, 'data')) as json_file:
+        data = json.load(json_file)
+        # Add system_info elements to our JSON
+        data["system_info"] = stable_baselines3.get_system_info(print_info=False)[0]
+
+    # Step 3: Write our config.json file
+    with open(Path(repo_local_path) / 'config.json', 'w') as outfile:
+        json.dump(data, outfile)
+
+
+def _evaluate_agent(model, eval_env, n_eval_episodes, is_deterministic, repo_local_path):
+    """
+    Evaluate the agent using SB3 evaluate_policy method and create a results.json
+    :param model: name of the model zip file
+    :param eval_env: environment used to evaluate the agent
+    :param n_eval_episodes: number of evaluation episodes (by default: 10)
```

Comment:
is it really 10 by default? Is that from `evaluate_policy`?

Reply:
No is from package_to_hub line 218 n_eval_episodes=10

Reply:
I would remove the "(by default: 10)" from here, since the default is in the function that calls this function, not here.